gnome-initial-setup/pages/live-chooser/gis-live-chooser-page.c

```diff
@@ -0,0 +1,305 @@
+                                       LIVE_ACCOUNT_USERNAME,
+                                       LIVE_ACCOUNT_FULLNAME,
+                                       ACT_USER_ACCOUNT_TYPE_ADMINISTRATOR,
+                                       &error);
+  if (error)
+    {
+      g_warning ("Failed to created shared user: %s", error->message);
+      g_clear_error (&error);
+      return;
+    }
+
+  act_user_set_password_mode (user, ACT_USER_PASSWORD_MODE_NONE);
+  act_user_set_automatic_login (user, FALSE);
+
+  language = gis_driver_get_user_language (GIS_PAGE (self)->driver);
+
+  if (language)
+    act_user_set_language (user, language);
+
+  gis_driver_set_user_permissions (GIS_PAGE (self)->driver, user, NULL);
+
+  g_object_unref (user);
+}
+
+static void
+user_loaded (GObject    *object,
+             GParamSpec *pspec,
+             gpointer    user_data)
+{
+  gchar *language = user_data;
+
+  act_user_set_language (ACT_USER (object), language);
+
+  g_free (language);
+}
+
+static void
+load_css_overrides (GisLiveChooserPage *self)
+{
+  GtkCssProvider *provider;
+  GError *error;
+  GFile *file;
+
+  error = NULL;
+  provider = gtk_css_provider_new ();
+  file = g_file_new_for_uri ("resource:///org/gnome/initial-setup/live-chooser.css");
+  gtk_css_provider_load_from_file (provider, file, &error);
+  g_object_unref (file);
+
+  if (error != NULL)
+    {
+      g_warning ("Unable to load CSS overrides for the live-chooser page: %s", error->message);
+      g_clear_error (&error);
+    }
+  else
+    {
+      gtk_style_context_add_provider_for_screen (gtk_widget_get_screen (GTK_WIDGET (self)),
+                                                 GTK_STYLE_PROVIDER (provider),
+                                                 GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+    }
+
+  g_object_unref (provider);
+}
+
+static void
+update_assistant (GisLiveChooserPage *self)
+{
+  GisAssistant *assistant = gis_driver_get_assistant (GIS_PAGE (self)->driver);
+
+  gis_assistant_next_page (assistant);
+}
+
+static void
+try_button_clicked (GisLiveChooserPage *self)
+{
+  create_live_user (self);
+  update_assistant (self);
+}
+
+static void
+installer_exited_cb (GPid     pid,
+                     gint     status,
+                     gpointer user_data)
+{
+  GisLiveChooserPage *self = user_data;
+  GError *error = NULL;
+
+  g_spawn_check_exit_status (status, &error);
+
+  if (error)
+    {
+      GtkWidget *message_dialog;
+
+      g_critical ("Error running the installer: %s", error->message);
+
+      message_dialog = gtk_message_dialog_new (GTK_WINDOW (gtk_widget_get_toplevel (user_data)),
+                                               GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+                                               GTK_MESSAGE_ERROR,
+                                               GTK_BUTTONS_CLOSE,
+                                               _("Error running the disk installer: %s"), error->message);
```

Comment:
why "disk" here but not elsewhere? I'd prefer no "disk", we don't call it the "disk installer" anywhere else.